packages/services/src/services/VideoPlayer/VideoPlayer.js

```diff
@@ -0,0 +1,190 @@
+      setTimeout(() => {
+        _scriptReady(resolve, reject);
+      }, 100);
+    } else {
+      reject();
+    }
+  } else {
+    _loadScript();
+    _scriptReady(resolve, reject);
+  }
+}
+
+/**
+ *
+ * Returns boolean if the _scriptLoading and _scriptLoaded flag is false
+ */
+function _loadScript() {
+  _scriptLoading = true;
+  const script = document.createElement('script');
+  script.src = _embedUrl;
+  script.async = true;
+  document.body.appendChild(script);
+
+  script.onload = function() {
+    _scriptLoaded = true;
+    _scriptLoading = false;
+  };
+}
+
+/**
+ *
+ * VideoPlayerAPI class with methods of checking script state and
+ * embed video meta data and api data
+ * ibm.com
+ */
+class VideoPlayerAPI {
+  /**
+   *
+   * Gets the full _scriptReady state
+   *
+   * @returns {Promise<*>} Promise kaltura video player file
+   */
+  static checkScript() {
+    return new Promise((resolve, reject) => {
+      _scriptReady(resolve, reject);
+    });
+  }
+
+  /**
+   * Gets the embed meta data
+   *
+   * @param {string} videoId  The videoId we're embedding the placeholder for.
+   * @param {string} targetId The targetId the ID where we're putting the placeholder.
+   * @returns {object}  object
+   */
+  static async embedVideo(videoId, targetId) {
+    return await this.checkScript().then(() => {
+      kWidget.embed({
+        targetId: targetId,
+        wid: '_' + _partnerId,
+        uiconf_id: _uiConfId,
+        entry_id: videoId,
+        flashvars: {
+          autoPlay: false,
+        },
+        // Ready callback is issued for this player:
+        readyCallback: function(playerId) {
+          var kdp = document.getElementById(playerId);
+          kdp.kBind('Play', function() {
+            return playerId;
+          });
+          var events = [
+            'layoutBuildDone',
+            'playerReady',
+            'mediaLoaded',
+            'mediaError',
+            'playerStateChange',
+            'firstPlay',
+            'playerPlayed',
+            'playerPaused',
+            'preSeek',
+            'seek',
+            'seeked',
+            'playerUpdatePlayhead',
+            'openFullScreen',
+            'closeFullScreen',
+            'volumeChanged',
+            'mute',
+            'unmute',
+            'bufferChange',
+            'cuePointReached',
+            'playerPlayEnd',
+            'onChangeMedia',
+            'onChangeMediaDone',
+          ];
+          for (var i = 0; i < events.length; i++) {
+            (function(i) {
+              kdp.kBind(events[i], function(event) {
+                return JSON.stringify(event);
+              });
+            })(i);
+          }
+        },
+      });
+    });
+  }
+
+  /**
+   * Gets the api data
+   *
+   * @param {string} videoId  The videoId we're embedding the placeholder for.
+   * @returns {object}  object
+   */
+  static async api(videoId) {
+    return await new kWidget.api({ wid: _partnerId }).doRequest(
```

Comment:
@mkothur should wrap this with the `checkScript` as well:
```
return await this.checkScript().then(() => {
   ... call kWidget.api and return the response...
}
```

Reply:
updated @jeffchew